src/lib/swapper/swappers/utils/helpers/helpers.ts

```diff
@@ -74,3 +81,21 @@ export const createEmptyEvmTradeQuote = (
     ],
   }
 }
+
+export const getTreasuryAddressForReceiveAsset = (assetId: AssetId): string => {
```

Comment:
Could this be by `ChainId` instead? Also instead of an exhaustive switch, we could have a mapping and simply throw if undefined

```typescript
export const DAO_TREASURY_BY_CHAIN_ID: Record<ChainId, string> = {
  [KnownChainIds.EthereumMainnet]: DAO_TREASURY_ETHEREUM_MAINNET,
  [KnownChainIds.OptimismMainnet]: DAO_TREASURY_OPTIMISM,
  [KnownChainIds.AvalancheMainnet]: DAO_TREASURY_AVALANCHE,
  [KnownChainIds.PolygonMainnet]: DAO_TREASURY_POLYGON,
  [KnownChainIds.GnosisMainnet]: DAO_TREASURY_GNOSIS,
  [KnownChainIds.BnbSmartChainMainnet]: DAO_TREASURY_BSC,
}
```

Reply:
i like this version @gomesalexandre , but would replace `ChainId` with `EvmChainId`, so as we add new  evm chains it will force us to keep this map up to date

Reply:
Excellent feedback. Updated accordingly in https://github.com/shapeshift/web/pull/4818/commits/0348c29fd253d00f8152957a948d872a9941a8cf.

I like that it now forces us to update the mapping when new `EvmChainId`s are added.